apps/dh/api-dh/source/DataHub.WebApi/Modules/ProcessManager/Requests/RequestOperations.cs

```diff
@@ -0,0 +1,154 @@
+// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Security.Claims;
+using Energinet.DataHub.ProcessManager.Orchestrations.Abstractions.Processes.BRS_023_027.V1.Model;
+using Energinet.DataHub.ProcessManager.Orchestrations.Abstractions.Processes.Shared.BRS_026_028;
+using Energinet.DataHub.WebApi.Clients.MarketParticipant.v1;
+using Energinet.DataHub.WebApi.Extensions;
+using Energinet.DataHub.WebApi.Modules.Common.Types;
+using Energinet.DataHub.WebApi.Modules.ProcessManager.Requests.Client;
+using Energinet.DataHub.WebApi.Modules.ProcessManager.Requests.Extensions;
+using Energinet.DataHub.WebApi.Modules.ProcessManager.Requests.Types;
+using MeteringPointType = Energinet.DataHub.Edi.B2CWebApp.Clients.v1.MeteringPointType;
+
+namespace Energinet.DataHub.WebApi.Modules.ProcessManager.Requests;
+
+public static class RequestOperations
+{
+    [Query]
+    [UsePaging]
+    [UseSorting]
+    // TODO: This query should be authorized
+    public static Task<IEnumerable<IActorRequestQueryResult>> GetRequestsAsync(
+        IRequestsClient client) => client.GetRequestsAsync();
+
+    [Query]
+    public static async Task<RequestOptions> GetRequestOptionsAsync(
+        IHttpContextAccessor httpContextAccessor,
+        IMarketParticipantClient_V1 marketParticipantClient)
+    {
+        // TODO: Create common functionality for this use case
+        var user = httpContextAccessor.HttpContext?.User;
+        var associatedActor = user?.GetAssociatedActor()
+            ?? throw new InvalidOperationException("No associated actor found.");
+
+        var selectedActor = await marketParticipantClient.ActorGetAsync(associatedActor);
+        return new RequestOptions(user, selectedActor.MarketRole.EicFunction);
+    }
+
+    public class RequestOptions(ClaimsPrincipal user, EicFunction marketRole)
```

Comment:
Comment: A class in a class, is this some hot chocolate magic?

Reply:
It is just a nested class, but I can extract it into Types if preferred 😄

Reply:
Please do :-)

Reply:
b17d3a874f81ebb11efca2460ec92a0e8077ec9e